### ISSUE TITLE
Go: handle version like beta1

### ISSUE THREAD

**Issue by mtanda** [aws/aws-lambda-builders#189]
**Description:**
`sam build` is failed with go1.15beta1.
```
$ go version
go version go1.15beta1 linux/amd64
```

**Steps to reproduce the issue:**
1. install go1.15beta1
2. execute `sam build`

**Observed result:**
Get error.
```
    version_parts = [int(x.replace("rc", "")) for x in out_parts[2].replace(self.LANGUAGE, "").split(".")]
ValueError: invalid literal for int() with base 10: '15beta1'
```

**Expected result:**
No error.

**Additional environment details (Ex: Windows, Mac, Amazon Linux etc)**
Ubuntu 18.04 (Windows/WSL)

**Comment by mtanda** [comment]
I think, validator need to remove string like `beta.*` in https://github.com/awslabs/aws-lambda-builders/blob/v1.0.0/aws_lambda_builders/workflows/go_modules/validator.py#L50

**Comment by hoffa** [comment]
Closing; resolved in 1.7.0.
